cmake/partition_manager.cmake

```diff
@@ -27,231 +21,298 @@ macro(add_region)
   endif()
 endmacro()
 
-get_property(PM_IMAGES GLOBAL PROPERTY PM_IMAGES)
-get_property(PM_SUBSYS_PREPROCESSED GLOBAL PROPERTY PM_SUBSYS_PREPROCESSED)
-
-if(EXISTS "${PM_STATIC_YML_FILE}" AND NOT IS_DIRECTORY "${PM_STATIC_YML_FILE}")
-  set(static_configuration_file ${PM_STATIC_YML_FILE})
-else()
-  set(static_configuration_file ${APPLICATION_SOURCE_DIR}/pm_static.yml)
+include(${ZEPHYR_BASE}/../nrf/cmake/extensions.cmake)
```

Comment:
I notices you are including the nrf specific extension file twice, and I understand we need it available in Zephyr as well.

I think we should take a closer look into cleaning this up, cause it is a bad practice to include a file multiple times.

Could you create a followup task, cause I don't think it should block this PR.

Reply:
I agree with you. I made an effort to fix this by including `nrf/cmake/extensions.cmake` from `zephyr/cmake/extensions.cmake` here https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/279

The TL;DR of why it's not merged is:
1 - The CI in fw-nrfconnect-zephyr is executed with the zephyr manifest file (or it was at that point at least), so the nrf repo is not there.
2 - The modules executed after the subsystems are included (in zephyr/CMakeLists.txt) making checking for existence hard.

I'll create a task.

Reply:
NCSDK-4984